### PR TITLE
Promote bf16 to f32.

### PR DESCRIPTION
**Pull request by dellis23**

### CODE REVIEW

**Comment by dellis23** [comment]
Let me know if there's a better way to organize this or tests you think are missing.

File: compiler/src/iree/compiler/InputConversion/MHLO/Passes.cpp [hunk 1]
```diff
@@ -35,6 +35,12 @@ static llvm::cl::opt<bool> clDemoteF64ToF32(
     llvm::cl::desc(
         "Converts all MHLO f64 ops and values into f32 counterparts."),
     llvm::cl::init(true));
+static llvm::cl::opt<bool> clPromoteBF16ToF32(
+    "iree-mhlo-promote-bf16-to-f32",
```
Comment (jpienaar) [inline compiler/src/iree/compiler/InputConversion/MHLO/Passes.cpp:39]: I think you are following naming convention here, but is there anything MHLO specific?

Comment (dellis23) [reply compiler/src/iree/compiler/InputConversion/MHLO/Passes.cpp:39]: No, just cargo culting.  Should I also move these to a different directory?

**Review by rsuderman — APPROVED** [review]

File: compiler/src/iree/compiler/InputConversion/MHLO/Passes.cpp (hunk 1 above)
Comment (jpienaar) [reply compiler/src/iree/compiler/InputConversion/MHLO/Passes.cpp:39]: Nope, that's good, it fits the flow, just checking.

**Comment by jpienaar** [comment]
Seems just clang-format pending, thanks

**Review by jpienaar — APPROVED** [review]

**Pull request merged**